File: java-example/src/test/java/ru/stqa/training/selenium/tests/CustomerRegistrationTests.java
package ru.stqa.training.selenium.tests;

import com.tngtech.java.junit.dataprovider.DataProviderRunner;
import com.tngtech.java.junit.dataprovider.UseDataProvider;
import org.junit.Test;
import org.junit.runner.RunWith;
import ru.stqa.training.selenium.model.Customer;

import java.util.Set;

import static org.junit.Assert.assertTrue;

@RunWith(DataProviderRunner.class)
public class CustomerRegistrationTests extends TestBase {

    @Test
    @UseDataProvider(value = "validCustomers", location = DataProviders.class)
    public void canRegisterCustomer(Customer customer) {
        Set<String> oldIds = app.getCustomerIds();

        app.registerNewCustomer(customer);

        Set<String> newIds = app.getCustomerIds();

        assertTrue(newIds.containsAll(oldIds));
        assertTrue(newIds.size() == oldIds.size() + 1);
    }

}

File: csharp-example/csharp-example/tests/CustomerRegistrationTests.cs
using System;
using System.Collections.Generic;
using NUnit.Framework;

namespace csharp_example
{
    [TestFixture]
    public class CustomerRegistrationTests : TestBase
    {
        [Test, TestCaseSource(typeof(DataProviders), "ValidCustomers")]
        public void CanRegisterCustomer(Customer customer)
        {
            ISet<String> oldIds = app.GetCustomerIds();

            app.RegisterNewCustomer(customer);

            ISet<String> newIds = app.GetCustomerIds();

            Assert.IsTrue(newIds.IsSupersetOf(oldIds));
            Assert.IsTrue(newIds.Count == oldIds.Count + 1);
        }
    }
}
